skbio/io/fasta.py

```diff
@@ -580,8 +598,10 @@ def _fasta_sniffer(fh):
             try:
                 list(zip(range(num_records),
                          _parse_fasta_raw(fh, _parse_quality_scores, 'QUAL')))
-            except FASTAFormatError:
+            except QualNonIntegerError:
```

Comment:
The sniffer can be refactored to read up to 10 seqs using `_fasta_to_generator` and check each seq's "sequence" (`.values`) to see if they look like quality scores.